File: Java_Chess/ChessBoard.java
class ChessBoard {

    private String[][] board; 

    public ChessBoard() {
        board = new String[8][8];
    }

    public void clearBoard() {
        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                board[i][j] = "nil";
            }
        }
    }

    public void setBoard() {
        System.out.println("Not implemented");
    }

    public void displayBoard() { 
        for(int i=0; i<8; i++) {
            for(int j=0; j<8; j++)
                System.out.print(board[i][j] + " ");
            System.out.println(" "); 
        }
    }
}

File: Ruby_Chess/ChessBoard.rb
class ChessBoard
    def initialize
      @board = Array.new(7){Array.new(7)}
    end

    def clear_board
      (0...7).each do |row|
        (0...7).each do |column|
          @board[row][column] = nil
        end
      end
    end

    def populate_board
      puts "Not implemented\n"
    end

    def display_board
      (0...7).each do |row|
        puts @board[row].inspect
      end
    end

    def is_real_position?(row, column)
      if row < 0 or row > 7
        return false
      elsif column < 0 or column > 7
        return false
      else
        return true
      end
    end

    def is_occupied?(row, column)
      @board[row][column] == nil ? true : false
    end

    def get_board
      @board
    end

    def update_board(piece, row, column)
      @board[row][column] = piece
    end
end